src/RpcClient/WalletAPI.cs

```diff
@@ -113,12 +114,12 @@ public Transaction ClaimGas(string key)
         /// </summary>
         /// <param name="keyPair">keyPair</param>
         /// <returns>The transaction sended</returns>
-        public Transaction ClaimGas(KeyPair keyPair)
+        public async Task<Transaction> ClaimGasAsync(KeyPair keyPair)
         {
             UInt160 toHash = Contract.CreateSignatureRedeemScript(keyPair.PublicKey).ToScriptHash();
-            BigInteger balance = nep5API.BalanceOf(NativeContract.NEO.Hash, toHash);
-            Transaction transaction = nep5API.CreateTransferTx(NativeContract.NEO.Hash, keyPair, toHash, balance);
-            rpcClient.SendRawTransaction(transaction);
+            BigInteger balance = await nep5API.BalanceOfAsync(NativeContract.NEO.Hash, toHash).ConfigureAwait(false);
```

Comment:
Maybe we can save the balance of, and transfer 0 tokens

Reply:
https://github.com/neo-project/neo/blob/6886991ca67f2c2148429ba87bf7d782f8cb12c8/src/neo/SmartContract/Native/Tokens/NeoToken.cs#L49

Reply:
I think we should limit this PR to focus on making the existing RpcClient async. Suggestions like this can be made in later PRs and after we ship preview 4.